### PR TITLE
change Jenkinsfile to use parallel declarative format

### PR DESCRIPTION
**Pull request by jakubgs**
This is a refactoring of current status-go build process to use Jenkins declarative pipelines and proper parallel builds.

I'm testing it here: https://ci.status.im/job/status-go/job/parallel/

Status: WIP

### CODE REVIEW

File: _assets/ci/Jenkinsfile [hunk 1]
```diff
@@ -1,79 +1,101 @@
...
+    stage('Release') { when { expression { params.GITHUB_RELEASE == true } }
+      steps {
+        sh "make prepare-release"
+        withCredentials([[
+          $class: 'UsernamePasswordMultiBinding',
+          credentialsId: 'status-im-auto',
+          usernameVariable: 'GITHUB_USER',
+          passwordVariable: 'GITHUB_TOKEN'
+        ]]) {
+          sh "yes | make release release_branch=${gitBranch}"
```
Comment (adambabik) [inline _assets/ci/Jenkinsfile:94]: I don't see any line that moves built files to a directory which is used by `github-release` triggered by this target.

File: _assets/ci/lib.groovy [hunk 2]
```diff
@@ -0,0 +1,44 @@
+def gitCommit() {
+  return GIT_COMMIT.take(6)
+}
+
+def timestamp() {
+  def now = new Date(currentBuild.timeInMillis)
+  return now.format('yyMMdd-HHmmss', TimeZone.getTimeZone('UTC'))
+}
+
+def sufix() {
```
Comment (pedropombeiro) [inline _assets/ci/lib.groovy]:
```suggestion
def suffix() {
```

File: _assets/ci/Jenkinsfile [hunk 3]
```diff
@@ -1,79 +1,101 @@
...
+            stage('Compile') { steps { dir(env.STATUS_PATH) {
+              sh 'make statusgo-android'
+            } } }
+            stage('Archive') { steps {
+              sh """
+                mv ${env.STATUS_PATH}/build/bin/statusgo-android-16.aar \
+                  ${env.WORKSPACE}/status-go-android-${lib.sufix()}.aar
+              """
+              sh 'ls -l'
+              archiveArtifacts "status-go-android-${lib.sufix()}.aar"
```
Comment (pedropombeiro) [inline _assets/ci/Jenkinsfile]: The call for iOS has parenthesis, we should choose one style

File: _assets/ci/Jenkinsfile [hunk 4]
```diff
@@ -1,79 +1,101 @@
...
+            stage('Archive') { steps {
+              sh """
+                mv ${env.STATUS_PATH}/build/bin/statusgo-ios-9.3-framework/status-go-ios.zip \
+                  ${env.WORKSPACE}/status-go-ios-${lib.sufix()}.zip
+              """
+              sh 'ls -l'
+              archiveArtifacts("status-go-ios-${lib.sufix()}.zip")
+            } }
+            stage('Upload') { steps { script {
+              lib.uploadArtifact( "status-go-ios-${lib.sufix()}.zip")
```
Comment (pedropombeiro) [inline _assets/ci/Jenkinsfile]:
```suggestion
              lib.uploadArtifact("status-go-ios-${lib.sufix()}.zip")
```

File: _assets/ci/Jenkinsfile (hunk 1 above)
Comment (jakubgs) [reply _assets/ci/Jenkinsfile:94]: Wait, isn't that supposed to be done by `make prepare-release`? That is there.

Comment (adambabik) [reply _assets/ci/Jenkinsfile:94]: Oh yeah. That's true. Sorry, I didn't notice `sh 'make prepare-release'`.

**Review by adambabik — APPROVED** [review]

**Review by pedropombeiro — APPROVED** [review]

**Pull request merged**